chain/neutrino.go

```diff
@@ -77,6 +77,7 @@ type NeutrinoClient struct {
 	isRescan   bool
 
 	clientMtx sync.Mutex
+	rescanMtx sync.Mutex
```

Comment:
Can you add a Godoc comment here describing what variables exactly this mutex is guarding?